spec/models/integration_collection_mappings/xero_collection_mapping_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe IntegrationCollectionMappings::XeroCollectionMapping, type: :model do
+  subject(:mapping) { build(:xero_collection_mapping) }
+
+  let(:mapping_types) do
+    %i[fallback_item coupon subscription_fee minimum_commitment tax prepaid_credit credit_note account]
+  end
+
+  it { is_expected.to define_enum_for(:mapping_type).with_values(mapping_types) }
+
+  describe '#external_id' do
+    let(:external_id) { SecureRandom.uuid }
+
+    it 'assigns and retrieve a setting' do
+      mapping.external_id = external_id
+      expect(mapping.external_id).to eq(external_id)
+    end
+  end
```

Comment:
I don't quite understand why this test is needed. It tests that the active record model has an attribute and knows how to set and get its value, that is it.

The same for the rest of the spec file and the xero_mapping_spec file below.

Reply:
Yes, but it's a jsonb attribute defined by `settings_accessors :external_id` so it tests if it is defined.